### DISCUSSION TITLE
Verify failed when use "make deploy-sepolia"

### DISCUSSION THREAD

**Discussion by meitedaf** [Cyfrin/foundry-full-course-cu#886] [Q&A]
### Issue Description:
I encountered an issue while attempting to verify the foundry-fund-me-f23 contract on the sepolia network. I have successfully deployed the contract, but I encountered the following error when trying to verify it:
```
Error: 
error sending request for url (https://api-sepolia.etherscan.io/api/?apikey=BX4I4MAZRAJPUIUZ797WNFZ7FDPH9RXPDF&module=contract&action=getabi&address=0x0d4cc6df9a5f0cb27d7cfc61deedde7f00c65ae3): error trying to connect: Connection reset by peer (os error 54)

Context:
- Error #0: error trying to connect: Connection reset by peer (os error 54)
- Error #1: Connection reset by peer (os error 54)
```
### Background:

I have successfully deployed the smart contract on the sepolia network without any issues.
The error seems to occur when attempting to communicate with the etherscan.io API.
### Attempted Solutions:

I have checked my network connection to ensure there are no interruptions or issues.(ping https://api-rinkeby.etherscan.io/api success)
I have also tried switching to a different network, but the problem persists.
I have verified the correctness of the API URL and parameters, ensuring there are no spelling errors or parameter issues.

**I am looking for advice on how to resolve this issue or if someone has encountered a similar problem and found a solution. Your assistance is greatly appreciated.**

**Comment by Jamster187** [comment]
Does your .env file contain the following variables?:

SEPOLIA_RPC_URL=url
MAINNET_RPC_URL=url
ETHERSCAN_API_KEY=apikey
PRIVATE_KEY=privkey

Also, where are you using "https://api-sepolia.etherscan.io/api/?apikey=BX4I4MAZRAJPUIUZ797WNFZ7FDPH9RXPDF&module=contract&action=getabi&address=0x0d4cc6df9a5f0cb27d7cfc61deedde7f00c65ae3"?

**Reply by meitedaf** [reply]
Yes, my .env file contain these variables. 
what's the meaning of **_where are you using "https://api-sepolia.etherscan.io/api/?apikey=BX4I4MAZRAJPUIUZ797WNFZ7FDPH9RXPDF&module=contract&action=getabi&address=0x0d4cc6df9a5f0cb27d7cfc61deedde7f00c65ae3"?_**
In the terminal, I input the command: _make deploy-sepolia_

**Reply by Jamster187** [reply]
Hey meitedaf,

I asked where that link came from because I didn't recognize the format as a URL we used, but it's simply being returned by the terminal so apologies for the confusion. I'm also trying to figure this out with you :)

After a bit of googling, it seems that a common cause of this error could be an invalid api key. You mentioned having checked that everything is correct - does your ETHERSCAN_API_KEY variable have a space between the equal sign and the key?

Good:

ETHERSCAN_API_KEY=BX4I4MAZRAJPUIUZ797WNFZ7FDPH9RXPDF

Bad:

ETHERSCAN_API_KEY = BX4I4MAZRAJPUIUZ797WNFZ7FDPH9RXPDF <--------- this wont work

Also, does your Makefile look like the following?:

```
-include .env

build:; forge build

deploy-sepolia:
	forge script script/DeployFundMe.s.sol:DeployFundMe --rpc-url $(SEPOLIA_RPC_URL) --private-key $(PRIVATE_KEY) --broadcast --verify --etherscan-api-key $(ETHERSCAN_API_KEY) -vvvvv
```
